### PR TITLE
Add RKE2 hardening guide and self assessment guide.

### PR DESCRIPTION
**Pull request by briandowns**
#### Proposed Changes ####

Documentation was needed to outline how to harden an RKE2 cluster as
well as a guide for users to perform self assessment against their
cluster.

#### Linked Issues ####

https://github.com/rancher/rke2/issues/84

### CODE REVIEW

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:124]: SP: documentaion > documentation

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:376]: SP?: windows_prefered_cluster > windows_preferred_cluster

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:42]: SP? RKE > RKE2 ?

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:55]: SP? RKE > RKE2 ?

File: docs/security_self_assessment.md
Comment (davidnuzik) [inline docs/security_self_assessment.md:13]: Not sure on this one - meant RKE2 or you were specifically referencing RKE?

File: docs/security_self_assessment.md
Comment (davidnuzik) [inline docs/security_self_assessment.md:15]: SP? "RKE nodes." > "RKE2 nodes." ?

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:5]: SP: restrictice > restrictive

Comment (davidnuzik) [reply docs/hardening_guide.md:5]: SP: wit > with

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:9]: Nit: This is kind of repeating line 3? I'm not going to cry about it - just noticed it. Not a big concern.

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:85]: Link?: "Additional information about CNI providers can be found here" missing link

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:87]: SP: "is provide below" > "is provided below"

Comment (davidnuzik) [reply docs/hardening_guide.md:87]: Nit: "Additional documentation about network policies can be found on the Kubernetes site." I think we should link to it - if don't know direct link at least link to kubernetes docs site landing page.

File: docs/hardening_guide.md
Comment (davidnuzik) [inline docs/hardening_guide.md:120]: Remove "the" before permissive

```suggestion
Execute this script to apply the default-allow-all.yaml permissive NetworkPolicy to all namespaces.
```

File: docs/security_self_assessment.md
Comment (davidnuzik) [inline docs/security_self_assessment.md:19]: SP? 
```suggestion
### 1 Master Node Security Configuration
```

**Review by davidnuzik — CHANGES_REQUESTED** [review]
Requesting some changes per my comments. Some are nits that should be taken with a grain of salt.

File: docs/hardening_guide.md
Comment (davidnuzik) [reply docs/hardening_guide.md:376]: Wasn't sure if this is a spelling mistake or if there is a bug in code and `windows_prefered_cluster` is actually expected.

File: docs/hardening_guide.md
Comment (briandowns) [reply docs/hardening_guide.md:9]: That's how it is in the orginal. I'm happy to remove it here though.

Comment (davidnuzik) [reply docs/hardening_guide.md:9]: It's a nit so I'll leave it to you. It's not a big deal just reading it felt weird because it repeats.

**Comment by briandowns** [comment]
Updated. Thanks for the feedback!

**Comment by davidnuzik** [comment]
Awesome! Thanks for taking the time to address the concerns I outlined. LGTM from a spelling/grammar/readability perspective. Now we should just need 1 LGTM from someone else -- I don't grok this on a technical level.

**Comment by cjellick** [comment]
I will review and merge this pr

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:7]:
```suggestion
This document is to be used by RKE2 operators, security teams, auditors and decision makers.
```

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:65]: I dont really know why we skipped 1.1.9 and 1.1.10 in the rancher guide, but I dont think we should skip it here.

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:73]: Why would this be not applicable? The etcd data directory is at `/var/lib/rancher/rke2/server/db/etcd` and should have permissions set to `700`.

I can see this is currently not the case with rke2 and it was likely lost in the script refactor and never picked up elsewhere.

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:75]: This remediation won't work for rke2. Its for rke1. our data dir is not at /var/lib/etcd. The 1.1.11.sh script assumes docker, which is not on the host.

**Comment by cjellick** [comment]
I think we have a pretty big gap between what is needed and what we have here. Much of this is only applicable to RKE1 and won't work for RKE2.

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:25]: I think all these config file controls _are_ applicable. We have them in rke2.

For example:
```
ls -al  /var/lib/rancher/rke2/agent/pod-manifests/kube-apiserver.yaml
-rw-r--r-- 1 root root 6692 Sep 23 03:15 /var/lib/rancher/rke2/agent/pod-manifests/kube-apiserver.yaml
```

File: docs/security_self_assessment.md
Comment (cjellick) [reply docs/security_self_assessment.md:65]: They are the tests for the CNI file perms and ownership.
I guess they are skipped because the are "manual" and it is hard to apply the control to RKE1.
But, we should still include the controls here with a remediation that explains that rke2 deploys canal as the cni by default using helm and thus there is is not explicit config file.

File: docs/security_self_assessment.md
Comment (cjellick) [reply docs/security_self_assessment.md:75]: I have additional related comments in the security-scan PR around this control

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:105]: This is applicable. My other comments from 1.1.11 apply here

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:137]: This is applicable. see https://github.com/rancher/security-scan/pull/22/files#r493592654

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:143]: This is applicable and RKE2 does store the admin kubeconfig on disk. See 1.1.13 above and/or https://github.com/rancher/security-scan/pull/22/files#r493597555

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:173]: It is applicable. See https://github.com/rancher/security-scan/pull/22/files#r493609155

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:189]: See comments for 1.1.19. Is applicable. dir location is different.

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:281]: script should be updated to use the actual location

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:275]: this should probably be updated to use rke2's location

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:304]: 1.2.1 has been dropped. Can we include it? We can test (and pass) it thusly:
```
ps -ef | grep kube-apiserver | grep -- --anonymous-auth | sed 's%.*anonymous-auth[= ]\([^ ]*\).*%\1%'
```

Comment (cjellick) [reply docs/security_self_assessment.md:304]: I see that we do include 1.2.1 in the security-scan pr

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:385]: example could/should be Node,RBAC

File: docs/security_self_assessment.md
Comment (cjellick) [inline docs/security_self_assessment.md:439]: I would drop the long list and just say `"not have "AlwaysAdmin" or ...`

File: docs/security_self_assessment.md
Comment (cjellick) [reply docs/security_self_assessment.md:65]: I followed up with Bill on his position around what controls we show/don't show

Going forward we are going to show all controls. If one doesn't apply to us, we have to explain in the remidation why.

File: docs/hardening_guide.md
Comment (cjellick) [inline docs/hardening_guide.md:1]: Much of this guide is specific to RKE1 deployment and not applicable to RKE2.

**Comment by cjellick** [comment]
Im taking a stab at these two docs.

File: docs/security_self_assessment.md
Comment (briandowns) [reply docs/security_self_assessment.md:25]: As I understand it, the pod manifest contains the cli args to pass and doesn't use a config file. The process itself is then looked at for the values further down.

**Pull request closed without merging**
